src/Compiler/Compiler.cpp

```diff
@@ -395,9 +395,8 @@ bool Compiler::RegisterCommands(TclInterpreter* interp, bool batchMode) {
   }
   if (m_simulator == nullptr) {
     m_simulator = new Simulator(m_interp, this, m_out, m_tclInterpreterHandler);
-  } else {
-    m_simulator->RegisterCommands(m_interp);
   }
+  m_simulator->RegisterCommands(m_interp);
```

Comment:
Didn't follow this one. Could you please explain this ?

Reply:
I guess this is a bug. It works for upstream because simulator initialized in the main.cpp. In Foedag initializing happened in that place. But because of this **if** statement RegisterCommands doesn't happened. So this change make sure the simulation commands are registered.